File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> st = new Stack<Integer>();
        int arr[] = new int[temperatures.length];
        for (int i = temperatures.length - 1;i>=0;i--)
        {
            while (st.isEmpty() == false && temperatures[st.peek()] <= temperatures[i])
            {
                st.pop();
            }
            if (st.isEmpty() == true)
            {
                arr[i] = 0;
            }
            else
            {
                arr[i] = st.peek() - i;
            }
            st.push(i);
        }
        return arr;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        arr = [0] * len(temperatures)
        for i in range(len(temperatures) - 1, -1, -1):
            count = 1
            # temp = []
            while len(stack) > 0 and temperatures[stack[-1]] <= temperatures[i]:
                # temp.append(stack.pop())
                stack.pop()
            if len(stack) > 0:
                arr[i] = stack[-1] - i
            # while temp:
            #     stack.append(temp.pop())
            stack.append(i)
        return arr